problemtools/verifyproblem.py

```diff
@@ -1460,14 +1471,135 @@ def validate(self, testcase: TestCase, submission_output: str) -> SubmissionResu
+                if content.startswith('<?xml') and '<svg' in content:
+                    return True
+            except Exception as e:
+                self.warning(f"Error checking SVG: {e}")
+        else:
+            return False        
+    
+    #Gets a path to the folder where it should save
+    #Then creates a new folder for this round of tests and copies over the file, then increments the folder counter by one
+    def save_image(self, file): 
+        save_folder_path = os.getcwd() + f"/saved_images/output-{self.counter}" #TODO works but get correct path  #TODO GET JUDGE NAME AND OUTPUT  from funbction call
+        os.makedirs(save_folder_path, exist_ok=True)
+        shutil.copy(file, save_folder_path)
+        self.counter = self.counter + 1
+
+    #Returns True if a visualizer exists
+    #Otherwise False and changes the local variable if this already has been raised               
+    def visualizer_exists(self)->bool: 
+        if not self._visualizer and not self._missing_visualizer: 
+            self._missing_visualizer = True
+            self.warning('No visualizer found')
+        return bool(self._visualizer)
+                        
+    def visualize(self, result_file: str, feedback_dir: str, context: Context): #TODO context istället för testcase för flaggan
+        res = []
+        
+        if not self.visualizer_exists():
+            return
+        
+        #Checks if there is only one validator and then selects it
+        #Otherwise raises warning 
+        if len(self._visualizer) == 1:  
+            visualizer = self._visualizer[0] 
+        else:
+            if self._has_warned_amount:
+                self._has_warned_amount = False
+                self.warning(f'Wrong amount of visualizer. \nExcpected: 1\nActual: {len(self._visualizer)}')
+            return
+
+        #Tries to run the visualzier and raises a warning if failed
+        try:  
+            status, runtime = visualizer.run(args=[result_file,feedback_dir])           
+            if status != 0:
+                self.warning(f'The output visualizer crashed, status: {status}')
+        except Exception as e:
+            self.warning(f'Error running output visualizer: {e}')
+        
+        # Iterates through all the files in the feedback directory and performs a file header check on all files with the allowed file extensions
+        file_extensions = [".png", ".jpg", ".jpeg", ".svg"]
+        for file in os.listdir(feedback_dir):
+            file = os.path.join(feedback_dir,file) #TODO Gör snyggare
+            for ext in file_extensions:
+                if file.endswith(ext):
+                    res.append(tuple((file, self.check_image_type(file))))
+       
+        if context.save_output_visualizer_images:  #If the flag was raised all images are saved
+            for i in range(len(res)):
```

Comment:
du kan använda python for loop
```python
for image_path, passed_validation in res:
    if passed_validation:
        self.save_image(image_path)

```

eller motsvarande om du tar bort passed_validation (please do)